core/src/mindustry/input/Placement.java

```diff
@@ -56,6 +56,30 @@
         return points;
     }
 
+    public static Array<Point2> followLineUntil(int startX, int startY, int endX, int endY){
+        Pools.freeAll(points);
+        points.clear();
+
+        Tile cursor = world.ltile(startX, startY);
+        Block filter = cursor.block();
+
+        while(true){
+            if(cursor.block() != filter) break;                        // keep looking until encountering a different block
+            if(points.contains(Pools.obtain(Point2.class, Point2::new).set(cursor.x, cursor.y))) break; // avoid biting its own tail
+            points.add(Pools.obtain(Point2.class, Point2::new).set(cursor.x, cursor.y));
+            if(cursor.x == endX && cursor.y == endY) break;            // do not look further than the cursor
+            cursor = cursor.getNearby(cursor.rotation());              // move cursor to next block to check
```

Comment:
I don't think blocks without rotation should be part of the line. Even if this works with walls, it's confusing and unintuitive.

Reply:
Agree with this, added a check to see if the block can rotate or not so only conveyors/conduits get changed